src/app/ReadHandler.h

```diff
@@ -379,6 +438,10 @@ class ReadHandler : public Messaging::ExchangeDelegate
     friend class chip::app::reporting::Engine;
     friend class chip::app::InteractionModelEngine;
 
+    // The report scheduler needs to be able to access StateFlag private functions IsReadHandlerReportable and IsChunkedReport to
```

Comment:
Again: I don't think it should need to touch IsChunkedReport(), as a followup.  That said, as a non-followup: I see no calls to IsChunkedReport() in the new code now.  There should in fact be some, in the ReadHandler, no?

For IsReadHandlerReportable(), this is fine for now, but we should consider making it public.

Reply:
Just realised my comment wasn't worded properly, it is IsReadHandlerReportable that is calling two private methods from the ReadHandler, I added those in the comment now.